### PR TITLE
Color Extraction based on Drain Photos

### PR DESCRIPTION
**Pull request by nagalamadaka20p**
This update allows users to upload images of their drains (less than 30kb, right now) and extract the main colors from the image. Users are then allowed to pick which color best represents their drain output and save it. Error messages will pop up if no file is entered. A loading icon appears when colors are being extracted. 

https://user-images.githubusercontent.com/77798750/182245004-5af07abf-782d-4698-921c-c55c2bcb5041.mp4

The image that was uploaded was this one:

flower1

### CODE REVIEW

**Comment by dyi1** [comment]
From the video, there seems to be a delay between clicking on the color and the color appearing, do you know why?

File: server/routes/datalogs.js [hunk 1]
```diff
@@ -4,11 +4,35 @@ const { NotFoundError } = require("../utils/error");
...
+  if (!req.body.photo) {
+    return res.status(400).send("No photo");
+  }
+  const { photo } = req.body;
+
+  const { id } = req.body;
+
+  const photoObject = new Parse.Object("Photo");
+
+  let drainPhoto = new Parse.File("photo.jpg", { base64: photo });
```
Comment (dyi1) [inline server/routes/datalogs.js:26]: Can this not be const?

**Review by dyi1 — COMMENTED** [review]
A lot of the added files seem to be from a package you downloaded. It seems to be for compression?

**Comment by dyi1** [comment]
Overall skipping all the random files the code looks good

File: server/routes/datalogs.js (hunk 1 above)
Comment (nagalamadaka20p) [reply server/routes/datalogs.js:26]: Yes it can be I can change this

**Comment by nagalamadaka20p** [comment]
I just tried clicking the colors again and it was an instantaneous change, not sure why the video showed a delay. Maybe I paused before clicking the color or something?

File: server/routes/datalogs.js (hunk 1 above)
Comment (nagalamadaka20p) [reply server/routes/datalogs.js:26]: Also I believe the added files are from got, which was a package I needed for using the color extraction API.

**Pull request merged**